fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneAnalyzerRegistryImpl.java

```diff
@@ -88,29 +89,63 @@ private LuceneAnalyzerRegistryImpl() {
         Pair<AnalyzerChooser, AnalyzerChooser> defaultAnalyzerChooserPair = getAnalyzerChooser(index, defaultAnalyzerName, type);
 
-        if (analyzerPerFieldName == null) {
-            return new LuceneAnalyzerCombinationProvider(defaultAnalyzerChooserPair.getLeft(), defaultAnalyzerChooserPair.getRight(),
-                    null, null);
-        }
-
         Map<String, AnalyzerChooser> indexAnalyzerChooserPerFieldOverride = new TreeMap<>();
         Map<String, AnalyzerChooser> queryAnalyzerChooserPerFieldOverride = new TreeMap<>();
 
-        LuceneIndexOptions.parseKeyValuePairOptionValue(analyzerPerFieldName).forEach((fieldName, analyzerName) -> {
-            Pair<AnalyzerChooser, AnalyzerChooser> perFieldAnalyzerChooserPair = getAnalyzerChooser(index, analyzerName, type);
-            indexAnalyzerChooserPerFieldOverride.put(fieldName, perFieldAnalyzerChooserPair.getLeft());
-            queryAnalyzerChooserPerFieldOverride.put(fieldName, perFieldAnalyzerChooserPair.getRight());
+        if (analyzerPerFieldName != null) {
+            LuceneIndexOptions.parseKeyValuePairOptionValue(analyzerPerFieldName).forEach((fieldName, analyzerName) -> {
+                Pair<AnalyzerChooser, AnalyzerChooser> perFieldAnalyzerChooserPair = getAnalyzerChooser(index, analyzerName, type);
+                indexAnalyzerChooserPerFieldOverride.put(fieldName, perFieldAnalyzerChooserPair.getLeft());
+                queryAnalyzerChooserPerFieldOverride.put(fieldName, perFieldAnalyzerChooserPair.getRight());
+            });
+        }
+
+        auxiliaryFieldInfo.forEach((fieldName, fieldInfo) -> {
+            addPerFieldAnalyzerIfNecessary(indexAnalyzerChooserPerFieldOverride, fieldName, fieldInfo, index);
+            addPerFieldAnalyzerIfNecessary(queryAnalyzerChooserPerFieldOverride, fieldName, fieldInfo, index);
         });
 
         return new LuceneAnalyzerCombinationProvider(defaultAnalyzerChooserPair.getLeft(), defaultAnalyzerChooserPair.getRight(),
                 indexAnalyzerChooserPerFieldOverride, queryAnalyzerChooserPerFieldOverride);
     }
 
+    private void addPerFieldAnalyzerIfNecessary(@Nonnull final Map<String, AnalyzerChooser> chooserPerFieldOverride,
+                                                @Nonnull final String fieldName,
+                                                @Nonnull final LuceneIndexExpressions.DocumentFieldDerivation fieldInfo,
+                                                @Nonnull final Index index) {
+        if (chooserPerFieldOverride.containsKey(fieldName)) {
+            // do not override already chosen field analyzer.
+            return;
+        }
+        if (isEligibleForNoOpAnalyzer(fieldInfo)) {
+            if (registry.isEmpty()
+                    || registry.get(LuceneAnalyzerType.FULL_TEXT) == null
+                    || registry.get(LuceneAnalyzerType.FULL_TEXT).get(ExactTokenAnalyzerFactory.NAME) == null) {
```

Comment:
Is this right if the user has specified that they're looking for the `AUTO_COMPLETE` analyzer?

Reply:
Good point! fixed it. Thanks!